### PR TITLE
[IoT] Add --tags to IoT Hub create and update

### PR DESCRIPTION
**Pull request by c-ryan-k**
**Description**

Added --tags parameters to `az iot hub create` and `az iot hub update` commands

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
IoT

**Comment by digimaun** [comment]
Approved with @c-ryan-k's intent to fix the recording conflicts.

**Review by digimaun — APPROVED** [review]

**Comment by yonzhan** [comment]
Please resolve the conflicting files first.

**Comment by c-ryan-k** [comment]
@yonzhan conflicting files (test recordings) have been fixed.

File: src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py [hunk 1]
```diff
@@ -404,7 +407,7 @@ def test_identity_hub(self, resource_group, resource_group_location, storage_acc
         location = resource_group_location
 
         private_endpoint_type = 'Microsoft.Devices/IoTHubs'
-        identity_hub = 'identity-test-hub-cli-1'
+        identity_hub = 'identity-test-hub-cli-2'
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py:410]: May I ask why `identity_hub` needs to be modified?
If it's not a reusable value, how about changing it to a random value?

Comment (c-ryan-k) [reply src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py]: @zhoxing-ms - sure, this value was only changed due to a previous test run failing to delete the resource.

I've gone ahead and changed the value to a randomly generated one and updated the test recording.

**Pull request merged**